projects/Inventaire/tests/basic.nix

```diff
@@ -0,0 +1,60 @@
+{
+  sources,
+  lib,
+  pkgs,
+  ...
+}:
+{
+  name = "Inventaire-basic";
+
+  nodes = {
+    machine =
+      {
+        config,
+        lib,
+        pkgs,
+        ...
+      }:
+      {
+        imports = [
+          (sources.inputs.nixpkgs + "/nixos/tests/common/x11.nix")
+          sources.modules.ngipkgs
+          sources.modules.services.inventaire
+          sources.examples.Inventaire.basic
+        ];
+
+        # couchdb + elasticsearch eats up memory
+        # leave some overhead for interactive firefox usage
+        virtualisation.memorySize = 4096;
+
+        environment.systemPackages = with pkgs; [
+          firefox
+        ];
+      };
+  };
+
+  # Need to see when terminals have launched
+  enableOCR = true;
+
+  testScript =
+    { nodes, ... }:
+
+    ''
+      start_all()
+
+      machine.wait_for_unit("inventaire.service")
+
+      # With ElasticSearch (or something compatible, like OpenSearch), we can actually expect full startup & check that it works
+      machine.wait_for_console_text("inventaire server is listening on port 3006")
```

Comment:
Is the service port configurable? If so we probably want an option for that and for opening up the firewall.

**PS:** we probably shouldn't hardcode the port, here.

Reply:
Port is part of the config that the user needs to supply, because the same config also houses all of the secrets to any involved services: https://codeberg.org/inventaire/inventaire/src/commit/bd11c3c6d8dd435caa4e63e67d93b31ace7a754e/config/default.cjs#L61

Reply:
How about we introduce a `secretsFile` option for those, then? Then the rest of the setting can be declaratively set in the module.